spec/controllers/users_controller_spec.rb

```diff
@@ -130,6 +130,55 @@
         expect(flash.now[:errors]).not_to be_nil
       end
     end
+
+    describe '#show_update_email' do
+      it 'renders the update email page' do
+        get :show_update_email, :params => { :user_id => user_id }
+
+        expect(response).to have_http_status(:success)
+        expect(subject).to render_template(:show_update_email)
+      end
+    end
+
+    describe '#update_email' do
+      it 'updates the user email address' do
+        stub_cognito_response(method: :admin_update_user_attributes, payload: {})
+        post :update_email, :params => { update_user_email_form: { email: "test1@test.com"}, user_id: user_id}
+        expect(subject).to redirect_to(update_user_path)
+      end
+
+      it 'fails with error when form not valid' do
+        stub_cognito_response(method: :admin_update_user_attributes, payload: {})
+        post :update_email, :params => { update_user_email_form: { email: ""}, user_id: user_id}
+        expect(response).to have_http_status(:bad_request)
+        expect(subject).to render_template(:show_update_email)
+      end
+
+      it 'fails with error when form not valid' do
+        form = UpdateUserEmailForm.new({ email: ""})
+        form.valid?
+        post :update_email, :params => { update_user_email_form: { email: ""}, user_id: user_id}
+        expect(response).to have_http_status(:bad_request)
+        expect(subject).to render_template(:show_update_email)
+        expect(flash[:errors]).to match(form.errors.full_messages.join(', '))
+      end
+
+      it 'fails with error when email already exists' do
+        stub_cognito_response(method: :admin_update_user_attributes, payload: 'AliasExistsException')
+        post :update_email, :params => { update_user_email_form: { email: "test@test.com"}, user_id: user_id}
+        expect(response).to have_http_status(:bad_request)
+        expect(subject).to render_template(:show_update_email)
+        expect(flash[:errors]).to match(t('users.update_email.errors.already_exists'))
+      end
+
+      it 'fails with error when a cognito error is thrown' do
+        stub_cognito_response(method: :admin_update_user_attributes, payload: 'ServiceError')
+        post :update_email, :params => { update_user_email_form: { email: "test@test.com"}, user_id: user_id}
+        expect(response).to have_http_status(:bad_request)
+        expect(subject).to render_template(:show_update_email)
+        expect(flash[:errors]).to match(t('users.update_email.errors.generic_error'))
+      end
+    end
```

Comment:
can we have a test where `update_user_email_form:` param is missing? and maybe one test where the email field is not present or the field is not called email?